contrib/babelfishpg_common/src/datetime.c

```diff
@@ -797,4 +804,540 @@ datetime_to_numeric(PG_FUNCTION_ARGS)
 	Timestamp timestamp_left = PG_GETARG_TIMESTAMP(0);
 	float8 result = calculateDaysFromDefaultDatetime(timestamp_left);
 	PG_RETURN_NUMERIC(DirectFunctionCall1(float8_numeric, Float8GetDatum(result)));
-}
\ No newline at end of file
+}
+
+/*
+ * Returns the difference of two timestamps based on a provided unit
+ * INT64 representation for bigints
+ */
+Datum
+timestamp_diff(PG_FUNCTION_ARGS)
+{
+
+	text        *field     = PG_GETARG_TEXT_PP(0);
+	Timestamp	timestamp1 = PG_GETARG_TIMESTAMP(1);
+	Timestamp	timestamp2 = PG_GETARG_TIMESTAMP(2);
+	int32 diff = -1;
+	int ok1;
+	int ok2;
+	int32 yeardiff;
+	int32 monthdiff;
+	int32 daydiff;
+	int32 hourdiff;
+	int32 minutediff;
+	int32 seconddiff;
+	int32 millisecdiff;
+	int32 microsecdiff;
+	struct pg_tm tt1,
+			   *tm1 = &tt1;
```

Comment:
To define a tt1, and make a pointer to tt1, looks weired, do we have a better way to define those ? And also, pls give a meaningful names instead of tm1, tm2 .